app/controllers/time_tracker_controller.rb

```diff
@@ -34,6 +34,8 @@ def stop
   end
 
   def update
+    start && return if @timer_session.nil?
```

Comment:
This will almost certainly error because you get multiple render/redirects. Generally is very bad practice to call one controller action from another.

Can you add a spec that described what you want the behaviour to be?